client/src/components/sources/sources.js

```diff
@@ -123,18 +115,11 @@ class Sources extends React.Component {
   }
 
   itemSelectChange(item) {
-    const { filteredItems } = this.state;
-
-    item.selected = !item.selected;
-    let selectedItems = filteredItems.filter(item => {
-      return item.selected === true;
-    });
-
-    this.setState({ selectedItems: selectedItems });
-
     Store.dispatch({
-      type: sourcesTypes.SOURCES_SELECTED,
-      selectedSources: selectedItems
+      type: item.selected
+        ? sourcesTypes.DESELECT_SOURCE
+        : credentialsTypes.SELECT_SOURCE,
```

Comment:
Looks like ```SELECT_SOURCE``` doesn't exist for ```credentialsTypes``` ... for testing I changed it to
```js
? sourcesTypes.DESELECT_SOURCE
: sourcesTypes.SELECT_SOURCE,
```

Reply:
copy/paste error.